File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {

        int[] bitsArray = new int[n + 1];

        for(int i = 0; i <= n; i++) {
            byte onesCounter = 0;
            int temp = i;
            while(temp > 0) {
                onesCounter += (temp % 2 == 0) ? 0 : 1;
                temp = temp / 2;
            }
            bitsArray[i] = onesCounter;
            onesCounter = 0;
        }
        
        return bitsArray;
    }
}

File: 0338-counting-bits/0338-counting-bits.js
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    var array = [];
    
    for(let i = 0; i <= n; i++) {
        let temp = i;
        array.push(temp.toString(2).split(' ').reduce((count, str) => count + (str.split('').filter(char => char === "1").length), 0));
        // console.log(`${temp} -> ${array}`);
    }

    return array;
};